autopush/metrics.py

```diff
@@ -92,21 +92,27 @@ def _make_tags(self, tags):
         return tags
 
     def increment(self, name, count=1, tags=None, **kwargs):
-        threads.deferToThread(
-            self._client.incr(self._prefix_name(name), count,
-                              tags=self._make_tags(tags))
+        reactor.callInThread(
```

Comment:
Why is `deferToThread` replaced with `callInThread`? `deferToThread` can be used to call a method in the threadpool according to its docs: https://twistedmatrix.com/documents/13.1.0/api/twisted.internet.threads.deferToThread.html

Reply:
Well, on inspection of the source, they both look like they do almost exactly the same thing. However, callInThread has no return value, while deferToThread returns a Deferred (basically a Promise or Future) which allows retrieving the called-in-thread function's return value.

In this case we don't need the return value, since we're just doing a fire and forget on some metrics. So we might as well not incur the gc overhead of creating/destroying those Deferreds since there's no return value from the metrics function.

Reply:
Pretty much what @fzzzy said. Plus `callInThread` also allowed more tests to pass without odd timing failures.